### PR TITLE
Make NewMeter's parameters non-optional

### PR DESCRIPTION
**Pull request by pattyshack**
Prep work for plumbing fully initialized meter parameters from block computer to
transaction invoker

### CODE REVIEW

**Review by SaveTheRbtz — APPROVED** [review]

File: fvm/meter/weighted_meter.go [hunk 1]
```diff
@@ -265,38 +312,19 @@ type WeightedMeter struct {
...
+func NewMeter(params MeterParameters) Meter {
 	m := &WeightedMeter{
-		computationLimit:       uint64(computationLimit) << MeterExecutionInternalPrecisionBytes,
-		memoryLimit:            uint64(memoryLimit),
-		computationWeights:     DefaultComputationWeights,
-		memoryWeights:          DefaultMemoryWeights,
+		computationLimit:       uint64(params.ComputationLimit) << MeterExecutionInternalPrecisionBytes,
+		memoryLimit:            params.MemoryLimit,
+		computationWeights:     params.ComputationWeights,
+		memoryWeights:          params.MemoryWeights,
```
Comment (SaveTheRbtz) [inline fvm/meter/weighted_meter.go:320]: maybe we can make params private and embed it instead? `MeterExecutionInternalPrecisionBytes` shift can be in `Default` and `WithComputationLimit` functions.

File: fvm/meter/weighted_meter.go [hunk 2]
```diff
@@ -246,6 +248,51 @@ func _() {
 	_ = x[int(common.MemoryKindLast)-len(DefaultMemoryWeights)-1]
 }
 
+type MeterParameters struct {
```
Comment (SaveTheRbtz) [inline fvm/meter/weighted_meter.go:251]: any reason for making fields public?

**Review by SaveTheRbtz — APPROVED** [review]

File: fvm/meter/weighted_meter.go (hunk 1 above)
Comment (pattyshack) [reply fvm/meter/weighted_meter.go:320]: i'll the fields private / embed the param, but will probably leave precision as a const for now

**Comment by pattyshack** [comment]
convert param fields to private, and embedded params into meter

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**